pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixSegmentOnlineOfflineStateModelGenerator.java

```diff
@@ -46,23 +40,37 @@ private PinotHelixSegmentOnlineOfflineStateModelGenerator() {
   public static final String DROPPED_STATE = "DROPPED";
   public static final String CONSUMING_STATE = "CONSUMING";
 
+  // Helix state transitions can be assigned a priority which is used by Helix
+  // to issue the order of state transitions. A lower priority value means a higher
+  // priority in Helix.
+  // Set all DROPPED related state transitions to have a higher priority.
+  // Handling DROPPED state transitions with a higher priority can help prevent
+  // servers from running into disk utilization problems.
+  public static final int DROPPED_STATE_TRANSITION_PRIORITY = 1;
```

Comment:
Let's also give higher priority to state transitions to `OFFLINE` state (e.g. 2) because that can reduce the memory pressure of the server.

Reply:
done